### DISCUSSION TITLE
Update Debian on apache/airflow docker container to Debian 11

### DISCUSSION THREAD

**Discussion by cksoest** [apache/airflow#21236] [Q&A] [answered]
Is there a way  to update Debian on de apache/airflow docker container to Debian 11. Because i want to extend the image with extra packages. To get the right version of the packages i need Debian 11 instead of 10. So is there a way to update Debain 10 to Debian 11.
thanks

**Comment by potiuk** [comment, marked as answer]
You can try to build your custom image as described in: https://airflow.apache.org/docs/docker-stack/build.html#building-from-pypi-packages. Not sure if it will work, maybe you will have to correct few other places or add another variables but it should be possible rather easily (in order to build custom image you need to anyhow checkout the Airlfow sources and you can modify the Dockerfile afterwards):

In this one - `buster` is used as a base and you might want to use "bullseye" if you want to use Debian 11 (but again it might need some more modifications).

```
docker build . \
    --build-arg PYTHON_BASE_IMAGE="python:3.7-slim-buster" \
    --build-arg AIRFLOW_VERSION="2.2.3" \
    --tag "my-pypi-selected-version:0.0.1"
```


We will likely attempt soon to migrate airflow to Debian 11 (we are refreshing the image for various reasons - Python 3.10 and ARM and it might be easier to do it with Debian 11. But no promises for that yet (we will have to see if there are some problems with that in our CI/tests). And it's likely to be out in 2.3.0

**Reply by cksoest** [reply]
I don't quite understand, is it possible for me to change the Debian version to 11. with dist-upgrade in the dockerfile or by modifing the default image source. Or something else.

**Reply by potiuk** [reply]
You need to build a new image yourself. This is how images work. You need to learn how to build images. Using dist-upgrade is certainly not recommended (you might try but this is a very bad antipattern as it will balloon the size of your image likely 2x times or more.

**Reply by potiuk** [reply]
Just read https://airflow.apache.org/docs/docker-stack/build.html#customizing-the-image - it describes all about custom image building.
